src/infrastructure/channels/reliable_channel.rs

```diff
@@ -0,0 +1,146 @@
+use super::Channel;
+
+use net::{LocalAckRecord, ExternalAcks, NetworkConfig, NetworkQuality, NetworkQualityMeasurer};
+use packet::header::{HeaderParser, HeaderReader, AckedPacketHeader, StandardHeader};
+use sequence_buffer::{SequenceBuffer, CongestionData};
+use infrastructure::{DeliveryMethod, Fragmentation};
+use error::{PacketErrorKind,NetworkResult};
+use packet::{Packet, PacketData,PacketTypeId};
+
+use std::io::{Cursor, Read};
+use std::time::Instant;
+
+/// This channel should be used for reliable processing of packets.
+///
+/// *Details*
+///
+///  1. Reliable.
+///  2. Guarantee of delivery.
+///  3. Option to specify if data needs to be ordered.
+///  4. Packets will not be dropped.
+///  5. Duplication not possible
+pub struct ReliableChannel {
+    // settings
+    ordered: bool,
+    config: NetworkConfig,
+
+    // reliability control
+    seq_num: u16,
+    waiting_packets: LocalAckRecord,
+    their_acks: ExternalAcks,
+    dropped_packets: Vec<Box<[u8]>>,
```

Comment:
I changed from `Packet` to box of bytes because packet also contained information not needed to be stored. When you request dropped packets we already know from which `addr` it came and we also don't need to know `delivery type` because we only do something with dropped packets if the packet type is `Reliable`

Reply:
Makes sense